2996_reflection/reflection.md

```diff
@@ -271,4 +183,202 @@ std::string name1 = name_of(^S);             // Also okay.
+}
+```
+:::
+
+This example also illustrates that bit fields are not beyond the reach of this proposal.
+
+
+## Fast Generation of Integer Sequence
+
+:::bq
+```c++
+#include <utility>
+#include <vector>
+
+template<typename T>
+consteval info make_integer_seq_refl(T N) {
+  std::vector args{^T};
+  for (T k = 0; k<N; ++k)  args.push_back(std::meta::reflect_value(k));
+  return substitute(^std::integer_sequence, args);
+}
+
+template<typename T>
+  using make_integer_sequence<typename T, T N> = [:make_integer_seq_refl<T>(N):];
+```
+:::
+
+
+## Enum to String
+
+One of the most commonly requested facilities is to convert an enum value to a string (this example relies on expansion statements [@P1306R1]):
+
+::: bq
+```c++
+template <typename E>
+  requires std::is_enum_v<E>
+constexpr std::string enum_to_string(E value) {
+  template for (constexpr auto e : std::meta::members_of(^E)) {
+    if (value == [:e:]) {
+      return std::string(std::meta::name_of(e));
+    }
+  }
+
+  return "<unnamed>";
+}
+
+enum Color { red, green, blue };
+static_assert(enum_to_string(Color::red) == "red");
+static_assert(enum_to_string(Color(42)) == "<unnamed>");
+```
+:::
+
+We can also do the reverse in pretty much the same way:
+
+::: bq
+```c++
+template <typename E>
+  requires std::is_enum_v<E>
+constexpr std::optional<E> string_to_enum(std::string_view name) {
+  template for (constexpr auto e : std::meta::members_of(^E)) {
+    if (name == std::meta::name_of(e)) {
+      return [:e:];
+    }
+  }
+
+  return std::nullopt;
+}
+```
+:::
+
+
+## Parsing Command-Line Options
+
+Our next example shows how command-line options could be automatically mapped to an "options" structure.
+For simplicity, we posit the existence of a range-like class type `ProgramArgs` that collects the traditional `(argc, argv)` parameters in a more friendly package.
+This example also uses an expansion statement for simplicity.
+
+::: bq
+```c++
+#include <sstream>
+#include <string>
+#include <meta>
+#include <ProgramArgs.h>
+
+template<typename Opts> bool parse_options(Opts *opts, ProgramArgs const &args) {
+  using namespace std;
+  using namespace std::meta;
+  bool success = true;
+  for (auto const arg = args.begin(); args != args.end(); ++args) {
+    template for(constexpr auto dm: members_of(^Opts, is_nonstatic_data_member)) {
+      if (arg->is_option_with_name(name_of(dm))) {
+        // Arg is of the form "--word" where "word" is the name of dm.
+        // Move to the option value, but remember the option tag:
+        auto const opt_arg = arg++;
+        if (arg == args.end()) {
+          cerr << "Option " << string(opt_arg) << " is missing a value." << endl;
+          success = false;
+          break;
+        }
+        using T = typename[:type_of(dm):];
+        if constexpr (requires (T d, istringstream is) { is >> d; }) {
+          T val;
+          istringstream is(string(*arg));
+          is >> val;
+          opts->[:dm:] = val;
```

Comment:
If we want to be complete, this part does need to be a runtime error:
```cpp
ispanstream iss(*arg);
if (is >> val) {
    opts.[:dm:] = val;
} else {
    // fail
}
```
or even:
```cpp
ispanstream iss(*arg);
if (is >> opts.[:dm:]; !is) {
    // fail
}
```